app/controllers/newflow/login_signup_controller.rb

```diff
@@ -0,0 +1,323 @@
+module Newflow
+  # Contains every action for login and signup
+  class LoginSignupController < ApplicationController
+    layout 'newflow_layout'
+    skip_before_action :authenticate_user!
+    before_action :newflow_authenticate_user!, only: [:profile_newflow]
+    before_action :save_new_params_in_session, only: [:login_form]
+    before_action :maybe_skip_to_sign_up, only: [:login_form]
+    before_action :store_authorization_url_as_fallback, only: [:login_form, :login, :signup_form, :signup]
+    before_action :known_signup_role_redirect, only: [:login_form]
+    before_action :restart_if_missing_unverified_user, only: [:verify_email, :verify_pin, :change_your_email]
+    before_action :exit_newflow_signup_if_logged_in, only: [:login_form, :signup_form, :student_signup_form, :welcome]
+    before_action :set_active_banners
+    before_action :cache_client_app, only: [:login, :welcome]
+    skip_before_action :check_if_password_expired
+    fine_print_skip :general_terms_of_use, :privacy_policy,
+                    except: [:profile_newflow, :verify_pin, :signup_done]
+
+    def login
+      handle_with(
+        AuthenticateUser,
+        success: lambda {
+          clear_newflow_state
+          sign_in!(@handler_result.outputs.user)
+          security_log(:sign_in_successful)
```

Comment:
True, older code always passes in `authentication_id`. I am not doing that because I don't **have** an `Authentication` as an `output` from the Lev Handler—reason why is because I don't **need** an Authentication anymore to log users in. The older code did, I believe. I could, of course, query for it but I don't think that's necessary. I may be wrong, please let me know. Keeping it as it is now because the user is always passed-in to the security log, so I don't think there's any data loss. I don't even think it's that helpful to be passing in an `authentication_id` (in the new flow, at least); in the old flow I think it was helpful so that we could find whether they logged in with a social auth or password.